### PR TITLE
enhance assess_deploy_lxd_profile.py

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description of change

Add new test cases to assess_deploy_lxd_profile.py
  Run with lxd-profile charm and subordinate lxd-profile charm
  Run with no lxd-profile charm and subordinate lxd-profile charm
  Do upgrades
  Run on a container with lxd-profile charm and subordinate lxd-profile charm

## QA steps

Setup for local acceptance testing with LXD
run: ./assess_deploy_lxd_profile.py lxd $GOPATH/bin/juju /tmp/artifacts testme

### CODE REVIEW

**Comment by hmlanigan** [comment]
There is a timing issue whereby finding machine profiles fails because the charm upgrade hasn't completed.  Suggestions on how to wait that the charm upgrade to complete are welcome!

**Comment by SimonRichardson** [comment]
> There is a timing issue whereby finding machine profiles fails because the charm upgrade hasn't completed. Suggestions on how to wait that the charm upgrade to complete are welcome!

I wonder if we can add a `upgrade-charm` hook that sets something in the notes/version, so that you can have a custom `wait_for` that does exactly what we want. It should be noted that `wait_for` just looks for changes in the status output and parses them.

**Comment by hmlanigan** [comment]
Resolved timing issue by adding a wait_for condition: WaitForLXDProfileCondition

File: acceptancetests/jujupy/wait_condition.py [hunk 1]
```diff
@@ -427,3 +428,37 @@ def total_seconds(self):
...
+        machine_info = dict(status.iter_machines())
+        machine = container = self.machine
+        if 'lxd' in machine:
+            # container = machine
+            machine = machine.split('/')[0]
+        try:
+            if 'lxd' in self.machine:
+                machine_lxdprofiles = machine_info[machine]['containers'][container]["lxd-profiles"]
+            else:
+                machine_lxdprofiles = machine_info[machine]["lxd-profiles"]
```
Comment (SimonRichardson) [inline acceptancetests/jujupy/wait_condition.py]: Ah that's even better! We can be sure that we've added these then, as we use LXD to get the information directly.

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**